### PR TITLE
lightgbm: add version v3.3.2

### PR DESCRIPTION
**Pull request by cyfdecyf**
LightGBM v3.3.3 ~ v3.3.5 are special release used to prevent R package from being archived on CRAN. So v3.3.2 is the actual latest release for now.

### CODE REVIEW

**Comment by cyfdecyf** [comment]
It will take some time before I have time to look into the clang build failure.

**Comment by cyfdecyf** [comment]
The clang build failed because cmake can't find openmp related library and headers. I confirmed this for installing openmp on Arch Linux.

Try installing libomp-dev for ubuntu_clang workflow and see if build can pass.

File: .github/workflows/ubuntu_clang.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
         run: |
           # TODO we will remove it later
           sudo apt-get update
-          sudo apt-get install -y libgl1-mesa-dev libglu1-mesa-dev clang libc++-dev p7zip
+          sudo apt-get install -y libgl1-mesa-dev libglu1-mesa-dev clang libc++-dev libomp-dev p7zip
```
Comment (waruqi) [inline .github/workflows/ubuntu_clang.yml:31]: 有 libomp 包，还有 openmp 包

Comment (cyfdecyf) [reply .github/workflows/ubuntu_clang.yml:31]: Ubuntu 上 clang 应该是用 libomp，安装 libgomp1 应该解决不了。另外我在手头的 Ubuntu 20.04 上倒是没找到 openmp。

Comment (waruqi) [reply .github/workflows/ubuntu_clang.yml:31]: add_deps("libomp") 不就好了，不是 libgomp1，里面也会去找系统 apt::libomp-dev

**Comment by waruqi** [comment]
可以了

**Pull request merged**